### ISSUE TITLE
N2D MIG secure boot failure bootloop after maintenance. 

### ISSUE THREAD

**Issue by fancybear-dev** [terraform-google-modules/terraform-google-container-vm#108]
### TL;DR

Using the N2D compute instance with confidential compute (forces us to use TERMINATE maintenance policy) and secure boot with Container Optimized OS (COS) in a MIG configuration, creates bootloops after the a machine is TERMINATED for maintenance. Secure boot fails after maintenance, as PCR 4 and PCR 7 are not in sync anymore.

Related docs;
https://cloud.google.com/compute/shielded-vm/docs/shielded-vm#linux
https://tianocore-docs.github.io/edk2-TrustedBootChain/release-1.00/3_TCG_Trusted_Boot_Chain_in_EDKII.html

### Expected behavior

That either the VM would still operate correctly, or that the MIG actually recreated the VM. 

### Observed behavior

The MIG sees the VM is unhealthy and says it will recreate, but it actually only updates the VM. The VM is in an hourly bootloop, where secure boot consistently fails. The only workaround found thus far, was a deployed of a new instance template -> so the MIG actually replaces the VM (along with all other healthy VMs).

### Terraform Configuration

```hcl
This is not specific to this module, but thought it would be suited here given the relevance to this module.
```

### Terraform Version

```sh
See above, not relevant.
```

**Comment by fancybear-dev** [comment]
Mirrored the issue here as well;

https://issuetracker.google.com/issues/269487070
